Contribute: Run a ZAP Hackathon (#1433)
zaproxy/zaproxy-website: site/content/docs/contribute/run-a-zap-hackathon.md

---
title: "Contribute: Run a ZAP Hackathon"
type: page
---

If you run a technical group like an [OWASP Local Chapter](https://owasp.org/chapters/) then how about running a ZAP
[Hackathon](https://en.wikipedia.org/wiki/Hackathon)?

> You can do it all on you own but if you reach out to the [ZAP Team](/docs/team/) (either directly or via

Reviewer: your own 
(With an r)

the [User Group](https://groups.google.com/group/zaproxy-users)) then we will be delighted to help you
as much as we can.

### Setup

Participants who want to change the ZAP code will need to set up a development environment - for that see the
[ZAP Developer Guide](/docs/developer/).

> You might like to suggest that those people set up this up prior to the event.

Reviewer: ```suggestion
You might like to suggest that those people set this up prior to the event.
```

### Easy Coding Tasks

Issues labelled 
[good first issue](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
should be a great place to start.
If those look too easy then move on to the ones labelled 
[good second issue](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+second+issue%22).

### Easy Non-Coding Tasks

Both the [documentation](../docs/) and [translations](../translate/) are great options for people who do not want to work on 
technical tasks.

### Scan Rules

If you have people with more security experience then helping to improve the [scan rules](../scan-rules) 

> could be an ideal task, and one that could be very collaborative,

Reviewer 2: ```suggestion
could be an ideal task, and one that could be very collaborative.
```

If you have some pentesters who are not so interested in coding then they could [test](../testing) ZAP against one of the many 
[deliberately vulnerable web applications](https://owasp.org/www-project-vulnerable-web-applications-directory/)). 

> They could then report any False Positives and False Negative they find to any people interest in coding, who

Reviewer 2: ```suggestion
They could then report any False Positives and False Negatives they find to any people interested in coding, who
```

could then look at improving the existing [scan rules](../scan-rules) or creating new ones.

If you focus on a specific app which we do not currently [regularly test ZAP against](/docs/scans/) then

> you could even work out how to automated ZAP to test and score this app and submit a PR to include it in that set!

Reviewer 2: ```suggestion
you could even work out how to automate ZAP to test and score this app and submit a PR to include it in that set!
```
